src/umpire/ResourceManager.cpp

```diff
@@ -91,6 +96,9 @@ ResourceManager::ResourceManager() :
   registry.registerMemoryResource(
       util::make_unique<resource::HostResourceFactory>());
 
+  registry.registerMemoryResource(
```

Comment:
Could we add a `registerInternalMemoryResource` so we can separate the resources?